provider/spark_test.go

```diff
@@ -53,18 +56,76 @@ func testMaterializeResource(store *SparkOfflineStore) error {
 	if !reflect.DeepEqual(fetchedTable, table) {
 		return fmt.Errorf("Did not properly register table")
 	}
-
+	testResourceMaterializationID := MaterializationID(fmt.Sprintf("%s/%s/%s", FeatureMaterialization, testResourceName, testResourceVariant))
 	materialization, err := store.CreateMaterialization(testResource)
 	if err != nil {
 		return err
 	}
-	fetchedMaterialization, err := store.GetMaterialization(materialization.ID())
+	fetchedMaterialization, err := store.GetMaterialization(testResourceMaterializationID)
 	if err != nil {
 		return err
 	}
 	if !reflect.DeepEqual(fetchedMaterialization, materialization) {
 		return fmt.Errorf("get materialization and create materialization return different results")
 	}
+	correctMaterialization := map[string]ResourceRecord{
+		"John Smith_0": ResourceRecord{"John Smith_0", reflect.ValueOf(35).Interface(), time.UnixMilli(int64(5))},
```

Comment:
Do you need int64 casting on the timestamp?

Reply:
Why do we need the reflect.valueof.interface?

Reply:
the parquet writer only stores timestamps as the int64 primitive types. I'm going to make a seperate PR that fixes all the annoying type stuff and manual conversion by storing the feature value data types as part of the primary table/feature table records, similar to how it is stored in our Redis implementation.